crates/forge_app/src/tool_executor.rs

```diff
@@ -27,12 +61,144 @@ impl<
         + ShellService
         + FollowUpService
         + ConversationService
-        + EnvironmentService,
+        + WorkflowService
+        + PolicyLoaderService
+        + EnvironmentService
+        + ConfirmationService
+        + AppConfigService,
 > ToolExecutor<S>
 {
     pub fn new(services: Arc<S>) -> Self {
         Self { services }
     }
+
+    /// Get policies, creating default ones if they don't exist
+    #[async_recursion::async_recursion]
+    async fn get_or_create_policies(
+        &self,
+        context: &mut ToolCallContext,
+    ) -> anyhow::Result<PolicyConfig> {
+        if let Some(policies) = self.services.read_policies().await? {
+            Ok(policies)
+        } else {
+            let mut app_config = self.services.read_app_config().await?;
+            if !app_config.should_create_default_perms.unwrap_or(true) {
+                return Ok(PolicyConfig::new());
+            }
+
+            match self
+                .services
+                .request_user_confirmation(
+                    TitleFormat::info(format!("No permissions policies found. Would you like to create a default policies file at {}", self.services.permissions_path().display())).to_string(),
+                    AddDefaultPoliciesResponse::iter().collect(),
+                ) {
+                Some(AddDefaultPoliciesResponse::Accept) => {
+                    self.services.init_policies().await?;
+                    context
+                    .send(crate::fmt::content::ContentFormat::Markdown(TitleFormat::info(format!(
+                    "Default policies file created at `{}`. You can always review and modify it as needed.",
+                    self.services.permissions_path().display()
+                    )).to_string()))
+                    .await?;
+                    self.get_or_create_policies(context).await
+                }
+                Some(AddDefaultPoliciesResponse::Reject) | None => {
+                    context.send(
+                        crate::fmt::content::ContentFormat::Markdown(TitleFormat::info(
+                            "Permissions policies not created. You will be prompted for permissions on each operation that requires them."
+                        ).to_string())
+                    ).await?;
+                    Ok(PolicyConfig::new())
+                },
+                Some(AddDefaultPoliciesResponse::RejectAndRemember) => {
+                    context.send(
+                        crate::fmt::content::ContentFormat::Markdown(TitleFormat::info(
+                            "Permissions policies not created. You will be prompted for permissions on each operation that requires them."
+                        ).to_string())
+                    ).await?;
+                    app_config.should_create_default_perms = Some(false);
+                    self.services.write_app_config(&app_config).await?;
```

Comment:
Can move this logic into `self.services.init_default_policy()` defined under policy service. this will make tool executor simpler.